pkg/clouduploader/operation.go

```diff
@@ -16,13 +16,34 @@ limitations under the License.
 
 package clouduploader
 
-import "github.com/aws/aws-sdk-go/service/s3/s3manager"
+import (
+	"io"
+	"strings"
+
+	"github.com/pkg/errors"
+
+	"github.com/aws/aws-sdk-go/service/s3/s3manager"
+	"gocloud.dev/blob"
+)
 
 const (
 	// backupDir is remote storage-bucket directory
 	backupDir = "backups"
 )
 
+const (
+	// Type of Key, used while listing keys
+
+	// KeyFile - if key is a file
+	KeyFile int = 1 << iota
+
+	// KeyDirectory - if key is a directory
+	KeyDirectory
+
+	// KeyBoth - if key is a file or directory
```

Comment:
What is this? how can it be both file and directory :-)

Reply:
@kmova it is the key to list both files and directory if we want. Lister gets the list all the content in a particular directory, we use these key to filter the files or directories or both.

Reply:
n_: ListKeyFile, ListKeyDirectory, ListKeyBoth